libs/gl-cli/src/node.rs

```diff
@@ -0,0 +1,327 @@
+        #[arg(long)]
+        retry_for: Option<u32>,
+        #[arg(long)]
+        maxdelay: Option<u32>,
+        #[arg(long)]
+        exemptfee: Option<u64>,
+        #[arg(long)]
+        localinvreqid: Option<Vec<u8>>,
+        #[arg(long)]
+        exclude: Option<Vec<String>>,
+        #[arg(long)]
+        maxfee: Option<u64>,
+        #[arg(long)]
+        description: Option<String>,
+    },
+    /// Stop the node
+    Stop,
+}
+
+pub async fn command_handler<P: AsRef<Path>>(cmd: Command, config: Config<P>) -> Result<()> {
+    match cmd {
+        Command::Log => log(config).await,
+        Command::GetInfo => getinfo_handler(config).await,
+        Command::Invoice {
+            label,
+            description,
+            amount_msat,
+            expiry,
+            fallbacks,
+            preimage,
+            cltv,
+            deschashonly,
+        } => {
+            invoice_handler(
+                config,
+                label,
+                description,
+                amount_msat,
+                expiry,
+                fallbacks,
+                preimage,
+                cltv,
+                deschashonly,
+            )
+            .await
+        }
+        Command::Pay {
+            bolt11,
+            amount_msat,
+            label,
+            riskfactor,
+            maxfeepercent,
+            retry_for,
+            maxdelay,
+            exemptfee,
+            localinvreqid,
+            exclude,
+            maxfee,
+            description,
+        } => {
+            pay_handler(
+                config,
+                bolt11,
+                amount_msat,
+                label,
+                riskfactor,
+                maxfeepercent,
+                retry_for,
+                maxdelay,
+                exemptfee,
+                localinvreqid,
+                exclude,
+                maxfee,
+                description,
+            )
+            .await
+        }
+        Command::Stop => stop(config).await,
+    }
+}
+
+async fn log<P: AsRef<Path>>(config: Config<P>) -> Result<()> {
+    let creds_path = config.data_dir.as_ref().join(CREDENTIALS_FILE_NAME);
+    let creds = match util::read_credentials(&creds_path) {
+        Some(c) => c,
+        None => {
+            return Err(Error::CredentialsNotFoundError(format!(
+                "could not read from {}",
+                creds_path.display()
+            )))
+        }
+    };
+
+    let scheduler = gl_client::scheduler::Scheduler::new(config.network, creds)
+        .await
+        .map_err(Error::custom)?;
+
+    let mut node: gl_client::node::Client = scheduler.node().await.map_err(Error::custom)?;
+    let mut stream = node
+        .stream_log(StreamLogRequest {})
+        .await
+        .map_err(Error::custom)?
+        .into_inner();
+
+    //TODO:  Use tokio::select!() and ctrl + c to cancel listening for log lines
+    loop {
+        tokio::select! {
+            biased;
```

Comment:
Not quite sure this is needed, since the ctrl_c signal is less likely to be ready when polled, and `biased` only applies to the cases where multiple branches are ready at the same time. It can prevent starvation of the `ctrl_c` branch though, which might have been your intention :thinking: 

Also the comment above is no longer valid.

Reply:
Nevermind, comment gets removed in 9df9d47adcd33c83fc386a9716159b94aef98777